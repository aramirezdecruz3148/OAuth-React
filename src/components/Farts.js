import React from 'react';
import { useAuth0 } from '../AuthOProvider';

export default function Farts() {
  const { isAuthenticated, loading, user } = useAuth0();
  return (
    <>
      <h3>{ isAuthenticated && !loading ? user.name : 'My Dog'} definitely dealt it!</h3>
    </>
  );
}
