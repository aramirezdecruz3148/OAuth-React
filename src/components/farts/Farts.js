import React from 'react';
import PropTypes from 'prop-types';

function Farts({ farts }) {
  const fartList = farts.map((fart, i) => (
    <li key={i}>
      <h5>{fart}</h5>
    </li>
  ));
  return (
    <ul>
      {fartList}
    </ul>
  );
}

Farts.propTypes = {
  farts: PropTypes.array.isRequired
};

export default Farts;
