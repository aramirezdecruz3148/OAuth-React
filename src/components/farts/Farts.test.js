import React from 'react';
import { shallow } from 'enzyme';
import Farts from './Farts';

describe('Farts component', () => {
  it('renders Farts', () => {
    const wrapper = shallow(<Farts farts={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
