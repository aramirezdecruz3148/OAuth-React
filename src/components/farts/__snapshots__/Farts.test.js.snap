// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Farts component renders Farts 1`] = `<ul />`;
